### DISCUSSION TITLE
Reducing css files

### DISCUSSION THREAD

**Discussion by pl71** [joomla/joomla-cms#43085] [2. Cold Features]
Having an [issue with the link color](https://github.com/joomla/joomla-cms/issues/43071#issuecomment-2003008978) found that
file `colors_alternative.css` is only 180 bytes long containing 5 CSS variables with global scope. 
@ReLater Is it possible for these variables to be written directly into `template.css`, thus reducing the number of files to be loaded?

**Comment by ReLater** [comment]
I think I'm the wrong contact partner here. If you want things like that in core (changing the colors scheme inside a template STYLE => rebuilding template.css files on-the-fly with selected colors scheme included) you need a JavaScript driven SASS compiler or so(?). This is beyond my capabilities.

Crux: If you have several template **style**s of the same template you must create a different template.css for each of them. One can use colors scheme x, the other y, the next a custom one... You need several CSS files then like template-87.css, template-99.css where the number is the templateStyleId or so.

Another possibility to reduce requests can be to put the colors_xyz.css code inline. But then you need additional PHP that reads the file content and includes the rules inline in the page.

Or using a tool (I think there are some on the Joomla extension market) that combine the CSS files and just load 1 CSS file.

However, I personally cannot judge the pros and cons as far as performance and such stories are concerned.
